api/v1alpha1/olsconfig_types.go

```diff
@@ -112,7 +112,20 @@ type DeploymentConfig struct {
 	// +kubebuilder:validation:Minimum=0
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Number of replicas",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:podCount"}
 	Replicas *int32 `json:"replicas,omitempty"`
-	// The resource requirements for OLS instance.
+	// API container settings.
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="API Container"
+	APIContainer APIContainerConfig `json:"api,omitempty"`
+	// Data Collector container settings.
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Data Collector Container"
+	DataCollectorContainer DataCollectorContainerConfig `json:"dataCollector,omitempty"`
+}
+
+type APIContainerConfig struct {
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Resource Requirements",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:resourceRequirements"}
+	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
+}
+
+type DataCollectorContainerConfig struct {
```

Comment:
@vbelouso We also want to configure this for our lightspeed-console-plugin container too.